src/main/java/com/evch/rrm/webserver/CustomWebServer.java

```diff
@@ -0,0 +1,288 @@
+        if (checkRequest(requestEntries)) {
+            if (requestEntries.get("get") != null) {
+                String response;
+                String source = requestEntries.get("get");
+                switch (source) {
+                    case "/api/time":
+                        response = """
+                                {
+                                    "currentServerTime": "%s"
+                                }""";
+                        sendTextInResponse(ContentTypes.JSON, String.format(response, LocalDateTime.now()), clientSocket);
+                        break;
+                    case "/api/stats":
+                        response = """
+                                {
+                                    "stats": "%s"
+                                }""";
+                        sendTextInResponse(ContentTypes.JSON, String.format(response, "Server has no stats"), clientSocket);
+                        break;
+                    default:
+                        ContentTypes contentType = ContentTypes.HTML;
+                        try {
+                            contentType = ContentTypes.getContentType(getSourceExtensionOrSlash(source));
+                        } catch (NoSuchElementException e) {
+                            send404(clientSocket);
+                        }
+                        int questionMarkIndex = source.indexOf("?");
+                        String sourceName = (questionMarkIndex == -1) ? source : source.substring(0, questionMarkIndex);
+                        sendFileInResponse(contentType, userDirStaticResources + (source.equals("/") ? "index.html" : sourceName), clientSocket);
+                }
+            }
+
+            if (requestEntries.get("post") != null) {
+                String response;
+                String source = requestEntries.get("post");
+                if (source.equals("/api/echo")) {
+                    response = """
+                            {
+                                "echoOutput": "Server saw your message: %s"
+                            }""";
+                    sendTextInResponse(ContentTypes.JSON, String.format(response,
+                            new JSONObject(requestEntries.get("body")).get("message")), clientSocket);
+                }
+            }
+        } else {
+            send404(clientSocket);
+        }
+        return !(requestEntries.containsKey("connection") && requestEntries.get("connection").contains("close"));
+    }
+
+    private boolean checkRequest(Map<String, String> headers) {
+        return (headers.containsKey("get") && headers.get("http_type").contains("http/1.1")
+                && headers.containsKey("host") && (headers.get("host").contains("localhost:8080") || headers.get("host").contains("localhost:8081"))
+                && headers.containsKey("user-agent") && headers.get("user-agent").contains("mozilla/5.0")
```

Comment:
don't get it. we allow only certain requests from certain browsers? 
Why hardcode 1 particular case? if the current host is 8082, then, it should not work?